packages/mds-db/index.ts

```diff
@@ -246,8 +246,8 @@ async function readDeviceIds(provider_id?: UUID, skip?: number, take?: number):
 }
 
 // TODO: FIX updateDevice/readDevice circular reference
-async function readDevice(device_id: UUID): Promise<Recorded<Device>> {
-  const client = await getReadOnlyClient()
+async function readDevice(device_id: UUID, useWriteable?: boolean): Promise<Recorded<Device>> {
```

Comment:
This feels like a bit of boolean plinko to me. It's not obvious when/why someone would want to use true. I have a suggestion. Instead of a boolean, why not use an optional `client` parameter instead; at least then when looking at it one might say "Oh, I already have a client; let me just reuse that". So below, then would be:
```
const client = await getWritableClient()
await readDevice(event_param.device_id, client)
```

Reply:
It doesn't seem like the business of the DB layer's clients to know about writeable vs readable clients. `getWriteableClient` and `getReadOnlyClient` aren't exported right now for that reason.

Reply:
That extra parameter is only currently used within db, but you're right. Whether the parameter is a boolean or a client instance, it exposes control over the client to the callers either way and that's probably not ideal. I think @janedotx makes a good point and perhaps what we really need is a public `readDevice` method and an internal implementation. `readDevice` and `writeEvent` could both call into the private method and pass the client.